psydac/fem/basic.py

```diff
@@ -114,6 +119,42 @@ def eval_field_gradient( self, field, *eta , weights=None):
 
         """
 
+    # ...
+    def __mul__(self, a):
+
+        if self.is_product:
+            spaces = list(self.spaces)
+        else:
+            spaces = [self]
+
+        if a.is_product:
+            spaces += a.spaces
+        else:
+            spaces += [a]
+
+        space = ProductFemSpace(*spaces)
+        if a.symbolic_space and self.symbolic_space:
+            space._symbolic_space =  self.symbolic_space*a.symbolic_space
+        return space
+
+    # ...
+    def __rmul__(self, a):
+        if a.is_product:
+            spaces = list(a.spaces)
+        else:
+            spaces = [a]
+
+        if self.is_product:
+            spaces += list(self.spaces)
+        else:
+            spaces += [self]
```

Comment:
Again, could we compress this code into two lines?

```
spaces = [*(   a.spaces if    a.is_product else    [a]),
          *(self.spaces if self.is_product else [self])]
```

Reply:
done